src/wasm-io.js

```diff
@@ -89,19 +89,31 @@ function newSlice(memory, alloc, data) {
  * @return {Uint8Array}
  */
 function getSliceData(memory, pointer, length) {
-  /**
-   * @param {Pointer} ptr
-   * @param {number} len
-   */
-  const getData = function*(ptr, len) {
-    const memView = new Uint8Array(memory.buffer);
-    for (let index = 0; index < len; index++) {
-      if (memView[ptr] === undefined) { throw new Error(`Tried to read undef mem at ${ptr}`); }
-      yield memView[ptr + index];
-    }
-  };
+  return new Uint8Array(memory.buffer, pointer, length);
```

Comment:
This gets rid of the "undefined memory access" check. Do you think it was unnecessary?

Reply:
I was mostly trying to avoid copying memory on return by just returning the array that's a view into memory. Although re-reading the documentation, I'm not sure my mental model was totally correct (like, maybe this is _still_ copying memory).